BonovoBluetooth/src/com/bonovo/bluetooth/BonovoBluetoothHandfree.java

```diff
@@ -359,49 +387,59 @@ protected void onResume() {
 			}
 		}
 		mDialpadPressCount = 0;
+		

```

Comment:
I need to test my latest changes in the car yet, but everything compiles and runs in the house.  

All the JNI code should be in place for call waiting, getting Bluetooth media information (track name, artist, current position, etc) and sending custom commands to the phone via SPP.

In the next day or two i'll try to get the call waiting service code connected to the UI and it should be looking pretty good.